### ISSUE TITLE
AKS Secure Baseline using vWAN vHUB

### ISSUE THREAD

**Issue by spotakash** [mspnp/aks-secure-baseline#205]
Is it possible to come up with vWAN vHUB based AKS Secure Baseline? I noticed current AKS Secure baseline is using traditional VNET HUB while newer implementation is using vWAN vHUB. Using vWAN HUB based implementation, we are finding challenge where outbound has to be defined either 'loadbalancer' or 'UDR'. 
Considering vWAN HUB (newer way of doing HUB SPOKE) propagate routing over vnet connection, we want to avoid allowing or create any UDR on spoke vnet (giving developer freedom to operate on SPOKE vnet but not to create any UDR/attached to subnet as those can overcome vHUB routing). https://docs.microsoft.com/en-us/rest/api/aks/managed-clusters/create-or-update#outboundtype

Basically, our design require outbound to go to AzFw sitting in other vnet so we create static route on vnet connection but if we use outbound as loadbalancer, it get bypassed. For UDR, we want to avoid. 

Any suggested in better way to address in vWAN HUB based implementation where all routing we control at vWAN HUB level.

**Comment by ckittel** [comment]
Yeah the choice to use traditional hub spoke is based on where a majority of customers are at the moment, built around what's most commonly deployed from a corporate networking perspective.  At the years roll by, I wouldn't be surprised if vWAN takes the cake there.  As for AKS baseline implementation with it, I don't think anyone on our team has tried it.  It's also not in the backlog currently to explore and give guidance on AFAIK.

I'd like to see our reference architecture be inclusive of this networking topology, providing some guidance around it (e.g. doable with these few changes, or not doable because $reason).  But presently we do not have the time to explore this.

Three options as I see it:
* Someone external to our team can explore and contribute to the reference architecture with that information.  We can even include a bit in the Reference Implementation (this repo) on it as well if it can be synthesized into an easy to consume alternative.
* Someone external to our team builds a derivative work from this that uses vWAN and its published as a standalone baseline alternative
* A prioritization conversation happens with @doodlemania2 and this becomes a team priority (replacing other scheduled work).

**Comment by spotakash** [comment]
Hey @ckittel  thanks for your inputs. We are discussing about it offline. Let's see how can we jointly work toward this. If AKS can learn route from vWAN HUB Connection and outbound_type support with additional option, it will be great.

**Comment by ckittel** [comment]
We are going to close the issue for now.  If @spotakash you make any progress, let us know.
